secret/secrets_manager_client_test.go

```diff
@@ -20,7 +20,7 @@ func TestSecretsManagerClient(t *testing.T) {
 	assert.Implements(t, (*SecretsManagerClient)(nil), &BasicSecretsManagerClient{})
 }
 
-func TestSecretsManagerCreateAndDeleteSecret(t *testing.T) {
+func TestSecretsManagerCRUDSecret(t *testing.T) {
 	checkAWSEnvVars(t)
 
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Since we're changing the testing structure to a for loop instead of a bunch of t.Runs, should I leave it as we had before `context.WithTimeout...` or still change it to `context.WithCancel(...)`?

Reply:
There are two contexts, the one inside the for loop and the one outside of it. This outer one (which applies to all tests), should just wrap `context.Background`, so it should use `context.WithCancel` and defer the cancel. The inner one (which applies to a single test) should use `context.WithTimeout` and wraps the outer context, so it looks ilke this:

```go
// Create one context that applies to all tests.
// All it does it cancel when the function ends, to ensure that
// anything that was running (like a hanging test) is stopped.
ctx, cancel := context.WithCancel(context.Background())
defer cancel()

...

for <stuff> {
    // Apply 30 second timeout for this test.
    // Each test gets 30 seconds to run before it gets cancelled for taking too long.
    tctx, tcancel := context.WithTimeout(ctx, 30 * time.Second)
    defer tcancel()
     <run test case>
}

Reply:
So the only instance where we use ctx is when defining tctx and tcancel?